ArduCopter/surface_tracking.cpp

```diff
@@ -101,3 +101,19 @@ void Copter::SurfaceTracking::set_surface(Surface new_surface)
     surface = new_surface;
     reset_target = true;
 }
+
+// initialise surface tracking by setting which surface to track
+void Copter::init_surf_track (void) {
+    switch (g.surf_track_mode)
```

Comment:
the curly braces should be on the same line as the "switch".

Reply:
I wonder how this initialisation interacts with the auxiliary switch initialisation.  We should add a comment to say which should be called first and/or perhaps disable the auxiliary switch initialisation and do it all in this method.  .. I'm not totally sure but we need to be sure that the two play well together.

Reply:
I wonder if @peterbarker has an opinion on whether this init should be within the SurfaceTracking class or not.  I suspect it should be although the parameter is in the vehicle class (as is the auxiliary switch) so it could theoretically go in either place.  I guess I'd recommend putting it in the SurfaceTracking class.

Reply:
> I wonder how this initialisation interacts with the auxiliary switch initialisation. We should add a comment to say which should be called first and/or perhaps disable the auxiliary switch initialisation and do it all in this method. .. I'm not totally sure but we need to be sure that the two play well together.

@rmackay9 Can we update the parameter on the basis of the physical state of auxiliary switch while its initialisation? Because we can update the value of parameter on the basis of state of auxiliary switch but can not change the physical state of switch on the basis of parameter value. This way it won't matter which is initialised first. We can prioritise switch to decide the surface I guess.

Reply:
> I wonder if @peterbarker has an opinion on whether this init should be within the SurfaceTracking class or not. I suspect it should be although the parameter is in the vehicle class (as is the auxiliary switch) so it could theoretically go in either place. I guess I'd recommend putting it in the SurfaceTracking class.

I tried to do the same earlier. But Surface tracking is a private member of copter class. And hence, we could not access it without a public method in copter. Please suggest me some way how should I do this.